.github/workflows/changelog.yml

```diff
@@ -15,19 +16,87 @@ jobs:
     if: ${{ !contains(github.event.pull_request.labels.*.name, 'dependencies') && !contains(github.event.pull_request.labels.*.name, 'Skip Changelog')}}
 
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.17
+      - name: Cache Go
+        id: go-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/bin
+            ~/go/pkg/mod
+          key: changelog-${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
 
-      - name: Check for CHANGELOG changes
+      - name: Ensure no changes to the CHANGELOG
         run: |
           # Only the latest commit of the feature branch is available
           # automatically. To diff with the base branch, we need to
           # fetch that too (and we only need its latest commit).
           git fetch origin ${{ github.base_ref }} --depth=1
           if [[ $(git diff --name-only FETCH_HEAD | grep CHANGELOG) ]]
           then
-            echo "A CHANGELOG was modified. Looks good!"
+            echo "The CHANGELOG should not be directly modified."
+            echo "Please add a .yaml file to the ./unreleased/ directory."
+            echo "See CONTRIBUTING.md for more details."
+            echo "Alternately, add the \"Skip Changelog\" label if this job should be skipped."
+            false
+          else
+            echo "The CHANGELOG was not modified."
+          fi
+
+      - name: Ensure one ./unreleased/*.yaml addition
```

Comment:
I think this question is unrelated to this PR but it made me think of it.

In the past there are situations when a PR needed multiple changelog entries.  Maybe there is a bug fix that also has a breaking change.  Can this new changelog process handle that?

Reply:
We could easily just allow multiple yaml files, but I think the vast majority of the time one entry is sufficient. Let's see how it goes and adjust as needed.

Reply:
I see we recently added a couple entries that require additional lines of text, so here's my proposal for that: #11863